platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/XMLUtils.java

```diff
@@ -223,6 +239,55 @@ private static void transformation(Source sourceXml, TransformerProperties trans
         return result.get();
     }
 
+    public DocumentBuilderFactory getSecureDocumentBuilderFactory(String className,
+            ClassLoader classLoader) {
+        DocumentBuilderFactory domFactory = DocumentBuilderFactory.newInstance(className,
+                classLoader);
+        setSecureSettings(domFactory);
+        return domFactory;
+    }
+
+    public DocumentBuilderFactory getSecureDocumentBuilderFactory() {
+        DocumentBuilderFactory domFactory = DocumentBuilderFactory.newInstance();
+        setSecureSettings(domFactory);
+        return domFactory;
+    }
+
+    public DocumentBuilder getSecureDocumentBuilder(boolean namespaceAware)
+            throws ParserConfigurationException {
+        DocumentBuilderFactory domFactory = getSecureDocumentBuilderFactory();
+        domFactory.setNamespaceAware(namespaceAware);
+        return domFactory.newDocumentBuilder();
+    }
+
+    public XMLReader getSecureXmlParser() throws SAXException {
+        XMLReader xmlParser = XMLReaderFactory.createXMLReader();
+        xmlParser.setFeature("http://xml.org/sax/features/external-general-entities", false);
+        xmlParser.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
+        xmlParser.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd",
+                false);
+        return xmlParser;
+    }
+
+    public Document parseDocument(InputStream inputStream)
```

Comment:
It would be nice to have a version of this method that allows the caller to make the parser namespace aware.